File: Lesson13_Active_Buzzer/code/c/Lesson13_Active_Buzzer.c
#include <lgpio.h>
#include <stdio.h>

#define BEEP_PIN  18  // BCM18 (wiringPi 1, physical pin 12)

int main(void)
{
    int h = lgGpiochipOpen(4);
    if (h < 0) {
        printf("Failed to open gpiochip4, try open gpiochip0\n");
        h = lgGpiochipOpen(0);
        if (h < 0) return -1;
    }

    lgGpioClaimOutput(h, 0, BEEP_PIN, 1);

    while (1) {
        lgGpioWrite(h, BEEP_PIN, 0);  // Beep on (active low)
        lguSleep(0.1);
        lgGpioWrite(h, BEEP_PIN, 1);  // Beep off
        lguSleep(0.1);
    }

    lgGpiochipClose(h);
    return 0;
}

File: Lesson13_Active_Buzzer/code/python/Lesson13_Active_Buzzer.py
#!/usr/bin/env python
from gpiozero import Buzzer
import time

# BCM18 (physical pin 12), active low buzzer
beep = Buzzer(18, active_high=False)

def setup():
    beep.off()  # Set pin high to turn off buzzer

def loop():
    while True:
        beep.on()
        time.sleep(0.1)
        beep.off()
        time.sleep(0.1)

def destroy():
    beep.off()
    beep.close()

if __name__ == '__main__':
    print('Press Ctrl+C to end the program...')
    setup()
    try:
        loop()
    except KeyboardInterrupt:
        destroy()
